sparrowpy/dsp.py

```diff
@@ -340,3 +339,146 @@ def energy_time_curve_from_impulse_response(
             signal.time[..., lower:upper]**2, axis=-1) * bw_factor
 
     return etc
+
+def band_filter_signal(signal:pf.Signal,
+                       frequencies:np.ndarray,
+                       num_fractions:int,
+                       order=4):
+    r"""Band filter input signal and return corresponding bandwidths.
+
+    This method filters an input signal into octave/3rd octave bands
+    with Butterworth filtering. By default, the filter order is set
+    as 4.
+    The user may input an arbitrary array of frequencies in
+    which to filter the signal. The closest fractional octave bands
+    to the input frequencies will be returned in the same order.
+
+    Returns the band-filtered signal in individual signal channels.
+
+    Parameters
+    ----------
+    signal: :py:class:`pyfar.Signal`
+        Input broad-spectrum signal.
+    frequencies: np.ndarray
+        Frequencies in which to filter the signal. In Hz.
+    num_fractions: int
+        The number of bands an octave is divided into. E.g., ``1`` refers to
+        octave bands and ``3`` to third octave bands.
+        All positive integers are allowed.
+    order: int
+        Butterworth filter order for the signal filtering. By default 4.
+
+    Returns
+    -------
+    band_filtered_noise: :py:class:`pyfar.Signal`
+        Band-filtered signal of cshape (signal.cshape + (len(freqs))
```

Comment:
```suggestion
        Band-filtered signal of cshape ``(signal.cshape + (len(freqs))``
```
is this really true? pyfar returns len(freq), signal.cshape?

Reply:
It wasn't true, I fixed it by swapping axes. This is important for the broadcasting in #128. Do you think we should comply with the pyfar convention?